sieves/tasks/predictive/ner/core.py

```diff
@@ -0,0 +1,221 @@
+            EngineType.glix,
+        }
+
+    def _validate_fewshot_examples(self) -> None:
+        for fs_example in self._fewshot_examples or []:
+            for entity in fs_example.entities:
+                if entity.entity_type not in self._entities:
+                    raise ValueError(f"Entity {entity.entity_type} not in {self._entities}.")
+
+    @property
+    def _state(self) -> dict[str, Any]:
+        return {
+            **super()._state,
+            "entities": self._entities,
+        }
+
+    def to_dataset(self, docs: Iterable[Doc]) -> datasets.Dataset:
+        """Convert NER results to a dataset.
+
+        :param docs: Documents with NER results.
+        :return: Dataset with NER results.
+        """
+        # Define metadata and features for the dataset
+        features = datasets.Features(
+            {
+                "text": datasets.Value("string"),
+                "entities": datasets.Sequence(
+                    {
+                        "text": datasets.Value("string"),
+                        "start": datasets.Value("int32"),
+                        "end": datasets.Value("int32"),
+                        "entity": datasets.Value("string"),
+                    }
+                ),
+            }
+        )
+
+        info = datasets.DatasetInfo(
+            description=f"Named Entity Recognition dataset with entity types {self._entities}. Generated with sieves "
+            f"v{Config.get_version()}.",
+            features=features,
+        )
+
+        # Fetch data used for generating dataset
+        try:
+            data = []
+            for doc in docs:
+                if self._task_id not in doc.results:
+                    raise KeyError(f"Document does not have results for task ID {self._task_id}")
+
+                # Get the entities from the document results
+                result = doc.results[self._task_id]
+                entities = []
+
+                # Handle different result formats
+                if hasattr(result, "entities"):
+                    # Pydantic model format
+                    for entity in result.entities:
+                        entities.append(
+                            {
+                                "text": entity.text,
+                                "start": entity.start,
+                                "end": entity.end,
+                                "entity": entity.entity_type,
+                            }
+                        )
+                elif isinstance(result, list):
+                    # List format (could be list of dictionaries or other entities)
+                    for entity in result:
+                        if isinstance(entity, dict) and "text" in entity:
+                            # Dictionary format with text key
+                            entity_dict = {
+                                "text": entity["text"],
+                                "start": entity.get("start", 0),
```

Comment:
Entities maybe having invalid data should be addressed in `consolidate()`, not here.